core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/BlockRequest.java

```diff
@@ -13,19 +13,36 @@
 package tech.pegasys.web3signer.core.service.http.handlers.signing.eth2;
 
 import tech.pegasys.teku.api.schema.BeaconBlock;
+import tech.pegasys.teku.api.schema.BeaconBlockHeader;
 import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.json.BlockRequestDeserializer;
 
+import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonDeserialize(using = BlockRequestDeserializer.class)
 public class BlockRequest {
   private final SpecMilestone version;
   private final BeaconBlock beaconBlock;
+  private final BeaconBlockHeader beaconBlockHeader;
```

Comment:
Doesn't the block header only apply for the bellatrix fork, it's not clear this optional. Perhaps we should at least make the getter return an Optional<BlockHeader> so it's clear. Probably not worth the effort to have a Bellatrix BlockRequest though?